src/Compilers/Core/Portable/Syntax/SyntaxToken.cs

```diff
@@ -504,6 +504,15 @@ public SyntaxToken WithTrailingTrivia(IEnumerable<SyntaxTrivia> trivia)
                 : default(SyntaxToken);
         }
 
+
+        /// <summary>
+        /// Creates a new token from this token with the trailing trivia removed.
+        /// </summary>
+        public SyntaxToken WithoutTrailingTrivia()
```

Comment:
>public SyntaxToken WithoutTrailingTrivia() [](start = 8, length = 42)

Do we have to add this API? It looks like WithTrailingTrivia does the job.

Reply:
@allisonchou can you move this to be an extension in the workspace layer.